to-do-task/src/components/add-task-form/AddTaskForm.js

```diff
@@ -23,10 +24,47 @@ export default function AddTaskForm({
   buttonName,
 }) {
   const [todo, setTodo] = React.useState(currentState || initialState);
```

Comment:
I think it's better if we extracted this logic to a custom hook  i.e `use-form`  that takes the `initialState` and a `config object` that has a `validation scheme` and returns the `currentState, errors, touched` along with all the form helpers `onChange, onBlur ...etc`

Reply:
you should also pass the `onSubmit handler` method to the `config` and the form should an `onSubmit callback that you pass to the form`

Reply:
this way we can have a generic hook that deals with our forms